src/storage/storage.cc

```diff
@@ -256,7 +259,14 @@ Status Storage::Open(DBOpenMode mode) {
     }
   }
 
-  std::shared_ptr<rocksdb::Cache> shared_block_cache = rocksdb::NewLRUCache(block_cache_size, -1, false, 0.75);
+  std::shared_ptr<rocksdb::Cache> shared_block_cache;
+
+  if (config_->rocks_db.block_cache_type == rocksdb::PrimaryCacheType::kCacheTypeLRU) {
+    shared_block_cache = rocksdb::NewLRUCache(block_cache_size, -1, false, 0.75);
```

Comment:
> nit: should we also use `rocksdb::LruCacheOptions` because `NewLRUCache` is deprecated?

thanks, fixed

Reply:
> because `NewLRUCache` is deprecated?

Oh maybe for now we have to still use `NewLRUCache` for being compatible with Speedb. Speedb currently does not offer the `LRUCacheOptions::MakeSharedCache` API like Rocksdb.